### PR TITLE
Add API for block commit

### PR DESCRIPTION
**Pull request by xiywang**
Signed-off-by: Xiyue Wang xiywang@redhat.com

### CODE REVIEW

File: virttest/qemu_monitor.py [hunk 1]
```diff
@@ -907,6 +907,31 @@ def block_stream(self, device, speed=None, base=None,
             cmd += " %s" % base
         return self.cmd(cmd)
 
+    def block_commit(self, device, speed=None, base=None, top=None,
+                     cmd="block_commit", correct=True):
+        """
+        Start block-commit job;
```
Comment (will-Do) [inline virttest/qemu_monitor.py]: no need ';' in the end

File: virttest/qemu_monitor.py [hunk 2]
```diff
@@ -907,6 +907,31 @@ def block_stream(self, device, speed=None, base=None,
...
+
+        :param device: device ID
+        :param speed: int type, lmited speed(B/s)
+        :param base: base file
+        :param top: top file
+        :param correct: auto correct command, correct by default
```
Comment (will-Do) [inline virttest/qemu_monitor.py]: So you missing the 'cmd' param here

File: virttest/qemu_monitor.py [hunk 3]
```diff
@@ -907,6 +907,31 @@ def block_stream(self, device, speed=None, base=None,
...
+
+        :return: The command's output
+        """
+        if correct:
+            cmd = self.correct(cmd)
+        self.verify_supported_cmd(cmd)
+        cmd += " %s" % device
+        if speed is not None:
```
Comment (will-Do) [inline virttest/qemu_monitor.py]: ```if speed``` should be enough

File: virttest/qemu_monitor.py [hunk 4]
```diff
@@ -1982,6 +2007,31 @@ def block_stream(self, device, speed=None, base=None,
...
+    def block_commit(self, device, speed=None, base=None, top=None,
+                     cmd="block-commit", correct=True):
+        """
+        Start block-commit job;
+
+        :param device: device ID
+        :param speed: int type, limited speed(B/s)
+        :param base: base file
+        :param top: top file
+        :param correct: auto correct command, correct by default
```
Comment (will-Do) [inline virttest/qemu_monitor.py]: And also here

**Comment by will-Do** [comment]
And for the commit msg, 'framework' is not appropriate, it's just an API actually

**Comment by will-Do** [comment]
BTW, seems some wrong may exist in your git config, you can see there are 3 participants in this pull request, please check the third one.

**Comment by xiywang** [comment]
@will-Do , fixed, please help to review again. thx

**Comment by xiywang** [comment]
@xutian , updated. Please help to review again.

**Comment by xutian** [comment]
Looks OK to me.

File: virttest/qemu_monitor.py [hunk 5]
```diff
@@ -907,6 +907,32 @@ def block_stream(self, device, speed=None, base=None,
             cmd += " %s" % base
         return self.cmd(cmd)
 
+    def block_commit(self, device, speed=None, base=None, top=None,
+                     cmd="block_commit", correct=True):
+        """
+        Start block-commit job
+
+        :param device: device ID
+        :param speed: int type, lmited speed(B/s)
```
Comment (balamuruhans) [inline virttest/qemu_monitor.py]: a typo "limited"

File: virttest/qemu_vm.py [hunk 6]
```diff
@@ -4283,6 +4283,20 @@ def block_stream(self, device, speed, base=None, correct=True):
         return self.monitor.block_stream(device, speed, base,
                                          cmd, correct=correct)
 
+    def block_commit(self, device, speed, base=None, top=None, correct=True):
+        """
+        start to commit block device, aka merge snapshot
+
+        :param device: device ID;
+        :param speed: limited speed, default unit B/s;
```
Comment (balamuruhans) [inline virttest/qemu_vm.py]: As @will-Do have mentioned, semicolon ";" not required

File: virttest/qemu_vm.py [hunk 7]
```diff
@@ -4283,6 +4283,20 @@ def block_stream(self, device, speed, base=None, correct=True):
...
+        :param base: base file;
```
Comment (balamuruhans) [inline virttest/qemu_vm.py]: same here

File: virttest/qemu_vm.py [hunk 8]
```diff
@@ -4283,6 +4283,20 @@ def block_stream(self, device, speed, base=None, correct=True):
...
+        :param top: top file;
```
Comment (balamuruhans) [inline virttest/qemu_vm.py]: same here

**Comment by xiywang** [comment]
@will-Do @xutian @balamuruhans , updated. Please help to review. Thx!

**Comment by xutian** [comment]
LGTM

**Pull request merged**